src/components/sidebar/linkView/LinkView.tsx

```diff
@@ -1,50 +1,73 @@
 import * as React from 'react';
-import { inject, observer } from 'mobx-react';
+import { observer } from 'mobx-react';
 import classnames from 'classnames';
+import { match } from 'react-router';
+import moment from 'moment';
 import ButtonType from '~/enums/buttonType';
-import TransitType from '~/enums/transitType';
-import { SidebarStore } from '~/stores/sidebarStore';
+import RouteService from '~/services/routeService';
+import { IRoutePathLink, IRoute } from '~/models';
+import RoutePathLinkService from '~/services/routePathLinkService';
 import { Checkbox, Dropdown, Button, TransitToggleButtonBar } from '../../controls';
 import InputContainer from '../InputContainer';
 import MultiTabTextarea from './MultiTabTextarea';
+import Loader from '../../shared/loader/Loader';
 import ViewHeader from '../ViewHeader';
 import * as s from './linkView.scss';
 
 interface ILinkViewState {
-    selectedTransitType: TransitType;
+    routePathLink: IRoutePathLink | null;
+    route: IRoute | null;
+    isLoading: boolean;
 }
 
 interface ILinkViewProps {
-    sidebarStore?: SidebarStore;
+    match?: match<any>;
 }
 
-@inject('sidebarStore')
 @observer
 class LinkView extends React.Component<ILinkViewProps, ILinkViewState> {
-    constructor(props: any) {
+    constructor(props: ILinkViewProps) {
         super(props);
         this.state = {
-            selectedTransitType: TransitType.BUS,
+            routePathLink: null,
+            route: null,
+            isLoading: true,
         };
     }
 
     public componentDidMount() {
-        if (this.props.sidebarStore) {
-            // TODO: fetch GraphSQL with linkId
-            // const linkId = this.props.sidebarStore!.openLinkId;
+        const routeLinkId = this.props.match!.params.id;
+        if (routeLinkId) {
+            this.fetchRoutePathLink(routeLinkId);
         }
     }
 
-    public componentWillUnmount() {
-        this.props.sidebarStore!.setOpenLinkId(null);
+    public componentWillReceiveProps(props: ILinkViewProps) {
+        const routeLinkId = props.match!.params.id;
+        if (routeLinkId) {
+            this.fetchRoutePathLink(routeLinkId);
+        }
     }
 
-    private toggleSelectedTransitType = (selectedTransitType: TransitType): void => {
-        this.setState({ selectedTransitType });
-    }
+    private async fetchRoutePathLink(id: string) {
+        const routePathLinkId = parseInt(id, 10);
+
+        this.setState({ isLoading: true });
+
+        const routePathLink =
+            await RoutePathLinkService.fetchRoutePathLink(routePathLinkId);
+
+        if (routePathLink) {
+            this.setState({ routePathLink });
 
-    private getFilters = () => {
-        return [this.state.selectedTransitType];
+            if (routePathLink.routeId) {
+                const route = await RouteService.fetchRoute(routePathLink.routeId!);
```

Comment:
We need route to get transit type, and route name, which are both visible in link view